rest/codegen/server.go

```diff
@@ -1,11 +1,288 @@
 package codegen
 
 import (
+	"fmt"
+	"text/template"
+
 	"goa.design/goa.v2/codegen"
+	goadesign "goa.design/goa.v2/design"
 	"goa.design/goa.v2/rest/design"
+	"net/http"
 )
 
+//todo: in encode headers and mappings
+//todo: in decode payload, query and paths
+
+const serverT = `
+{{define "handlers_struct" -}}
+// {{.Name}}Handlers lists the {{.Name}} service endpoint HTTP handlers.
+type {{.Name}}Handlers struct {
```

Comment:
I was thinking that instead of doing the goifys and transforms inline (and potentially forgetting to do them like here) it would be better to build an intermediary data structure that contains all the values that the template can readily use. That was the idea behind the `data` field of the writers. So instead of passing the resource directly as data we should build a data structure that is convenient for the template (i.e. identify the variables that the template requires and compute their value in code then build the input data structure with the results).